src/logging.cc

```diff
@@ -961,6 +969,78 @@ bool LogFileObject::CreateLogfile(const string& time_pid_string) {
   return true;  // Everything worked
 }
 
+void LogFileObject::DeleteIfMaxNum(string &strDir)
+{
+	if (strDir.empty()) return;
+	string strSeverityName(GetLogSeverityName(severity_));
+	int fileMaxSizeNum = 0;
+#if defined(OS_WINDOWS)
+	long   hFile   =   0;
+	struct _finddata_t *earliestFile = (struct _finddata_t *) malloc(sizeof(struct _finddata_t));
```

Comment:
Is there a reason for heap allocation of `_finddata_t`?

Reply:
sorry, i  misunderstand the struct of _finddata_t, for it may too large while out of the function stack

Reply:
ok i will change this, thankyou